sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/V2PredicateSuite.scala

```diff
@@ -287,7 +287,7 @@ class V2PredicateSuite extends SparkFunSuite {
       new Predicate("=", Array[Expression](ref("a"), LiteralValue(1, IntegerType))),
       new Predicate("=", Array[Expression](ref("b"), LiteralValue(1, IntegerType))))
     assert(predicate1.equals(predicate2))
-    assert(predicate1.references.map(_.describe()).toSeq == Seq("a", "b"))
+    assert(predicate1.references.map(_.describe()).toSeq.sorted == Seq("a", "b"))
     assert(predicate1.describe.equals("(a = 1) OR (b = 1)"))
```

Comment:
We are sure there is no ordering expected in the rest of the code right ?
Given this test, I want to make sure we are not missing anything (the test could simply have been an artifact of existing sorted order ofcourse)

Reply:
@huaxingao already helped to check that order is not important, but some override in subclass is keep order.

https://github.com/apache/spark/blob/a2304e45d31d3d802a3976b9cd1ea0f72a2d604b/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala#L109-L111

Personally, I still hope to revert to the previous orderly version ....

Reply:
@mridulm Do you have any suggestions?

Reply:
I might have missed @huaxingao's comment - if she is fine with it, looks good to me :-)